test/e2e/storage/volume_provisioning.go

```diff
@@ -839,7 +839,7 @@ var _ = utils.SIGDescribe("Dynamic Provisioning", func() {
 	Describe("DynamicProvisioner Default", func() {
 		It("should create and delete default persistent volumes [Slow]", func() {
 			framework.SkipUnlessProviderIs("openstack", "gce", "aws", "gke", "vsphere", "azure")
-
+			framework.SkipUnlessDefaultStorageClassPresent(c, false)
```

Comment:
Other tests like PVC protection actually fail if there's no default storageclass. Maybe it should be updated to skip as well.  I think there are a few more test cases that currently require default storageclass, and will have to be sweeped.

Reply:
Yeah, right.
And that was a good chance to know I needed to configure default storageclass in my environment by investigating the test failure reason.
So I feel this kind of automatic skip has both benefit and drawback.

Reply:
Thanks, you bring up some good points.  What do you think of this idea?

* Update all tests that require default storageclass to check for the existence of one
* Tag all those tests consistently so that environments that don't support a default storageclass can skip the tests with that tag.

This way, by default tests will fail and notify that one should be installed.  Or you can explicitly choose to skip the tests.

Reply:
Sorry for my late response.

I cannot follow "Update all tests that require default storageclass to check for the existence of one" means. Does it each test which requires default storageclass check the existance of default one and automatically skip if not existing?

The second idea (tag all default storageclass consistently) seems good for me because testers can control tests based on their environments.

Reply:
Sorry this fell off my radar.  I meant that all storage e2e tests we have today that use the default storage class should share the same logic that checks that one is installed and fails if it doesn't.